pkg/kstatus/observe/aggregator/aggregator.go

```diff
@@ -0,0 +1,73 @@
+package aggregator
+
+import (
+	"sigs.k8s.io/cli-utils/pkg/kstatus/observe/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/wait"
+)
+
+// AllCurrentOrNotFoundStatusAggregator is an implementation of the StatusAggregator interface
+// that considers the aggregate status as Current whenever all resources have either the Current
+// or the NotFound status.
+// TODO: Treating resources that doesn't exist as Current is weird. But it kinda does
+// make sense when we track resources that are deleted/pruned. We should see if there is a better
+// way to handle this.
+type AllCurrentOrNotFoundStatusAggregator struct {
```

Comment:
Per the PR description, lets just call this `BasicAggregator` for now.

Please shorten the description, e.g. 

```
BasicAggregator implements StatusAggregator.
Aggregate status will be Current when all observed
resources are either Current or NotFound.
```